### PR TITLE
Show comments position

### PR DESCRIPTION
**Pull request by IanPhilips**
If a user has made a recent bet, they'll be prompted to comment on it. Otherwise, they'll be commenting with their position.

Ungrouped bets per user request in discord in the bets tab.

Added a tooltip for free response answer labels when a user comments as they can be unreadable when long.

### CODE REVIEW

File: web/components/feed/feed-items.tsx [hunk 1]
```diff
@@ -244,18 +311,68 @@ export function CommentInput(props: { contract: Contract }) {
...
+  noFloorShares = Math.floor(noShares)
+
+  const userPosition = yesFloorShares || noFloorShares
+  const { saleValue } = calculateCpmmSale(
+    contract as FullContract<CPMM, Binary>,
+    yesShares || noShares,
+    yesFloorShares > noFloorShares ? 'YES' : 'NO'
+  )
+  const userPositionMoney = formatMoney(Math.abs(saleValue))
+  return [userPosition, userPositionMoney, yesFloorShares, noFloorShares]
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:360]: Style nit: For many values, it's better to return an object (using '{}') than a tuple.

File: web/components/feed/feed-items.tsx [hunk 2]
```diff
@@ -206,14 +237,50 @@ export function CommentInput(props: { contract: Contract }) {
     setComment('')
   }
 
+  // Should this be oldest bet or most recent bet?
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:240]: I like most recent! Feel free to leave the comment though.

File: web/components/feed/activity-items.ts [hunk 3]
```diff
@@ -308,6 +311,37 @@ function groupBetsAndComments(
...
+    return bets.reduce((acc, bet) => {
+      const userId = bet.userId
+      if (!acc[userId]) {
+        acc[userId] = []
+      }
+      acc[userId].push(bet)
+      return acc
+    }, {} as { [userId: string]: Bet[] })
+  }
+  const betsByUserId = mapBetsByUserId(bets)
```
Comment (jahooma) [inline web/components/feed/activity-items.ts:329]: One suggestion is to use `_.groupBy(bets, bet => bet.userId)`. It's a bit easier!

**Review by jahooma — APPROVED** [review]
Sweet! Thanks for all this — no major complaints :-)

One overall comment is that it's a bit better to break up PR's if you can into independent pieces. Np though if it would be too much extra work. I've been there.

**Comment by IanPhilips** [comment]
I think I had too many commits here because I tried too many different ways of doing things with some of them looking rather bad. Should've made more mocks before coding it, I was too confident it would look good!

**Pull request merged**